heroic-component/src/main/java/com/spotify/heroic/metric/QueryResultPart.java

```diff
@@ -62,8 +65,18 @@
                 .map(ResultGroup.toShardedResultGroup(shard))
                 .iterator());
 
+            long sampleSize = 0;
+            if (result.getStatistics().getCounters().containsKey(AggregationInstance.SAMPLE_SIZE)) {
```

Comment:
This would probably be smoother if you add an accessor to Statistics:
```
Optional<Long> Statistics#getCounterValue(String key)
```

Then this would become:
```java
final long sampleSize = result.getStatistics().getCounterValue(AggregationInstance.SAMPLE_SIZE).orElseGet(() -> {
  // calculate over groups
});
```